docs/style-guide.md

```diff
@@ -0,0 +1,291 @@
+        Self {transform,transform_matrix,origin,matrix,dirty,logger}
+    }
+}
+
+// === Getters ===
+
+impl<OnChange> HierarchicalTransform<OnChange> {
+    pub fn position(&self) -> &Vector3<f32> {
+        &self.transform.position
+    }
+
+    pub fn rotation(&self) -> &Vector3<f32> {
+        &self.transform.rotation
+    }
+
+    ...
+}
+
+// === Setters ===
+
+impl<OnChange:Callback0> HierarchicalTransform<OnChange> {
+    pub fn position_mut(&mut self) -> &mut Vector3<f32> {
+        self.dirty.set();
+        &mut self.transform.position
+    }
+
+    pub fn rotation_mut(&mut self) -> &mut Vector3<f32> {
+        self.dirty.set();
+        &mut self.transform.rotation
+    }
+
+    ...
+}
+```
+
+
+### Vertical spacing
+We use the following amount of vertical spacing:
+- 2 blank lines after imports
+- 2 blank lines before each section.
+- 1 blank line before and after sub-section.
+- 1 blank line before functions / structures / impls.
+- 1 blank line at the end of the file.
+
+
+### Multiline Expressions
+Most (preferably all) expressions should be single line. Multiline expression is
+hard to read and introduces noise in the code. Often, it is also an indicator of
+a code which is not properly refactored. Try to refactor parts of multiline 
+expressions to well-named variables, and divide them to several single-line 
+expressions.
+
+Example of bad formatted code:
+```rust
+pub fn new() -> Self {
+    let shape_dirty = ShapeDirty::new(logger.sub("shape_dirty"),
+        on_dirty.clone());
+    let dirty_flag = MeshRegistryDirty::new(logger.sub("mesh_registry_dirty"),
+        on_dirty);
+    Self { dirty_flag, dirty_flag }
+}
+```
+Example of properly formatted code:
+
+```rust
+pub fn new() -> Self {
+    let sub_logger  = logger.sub("shape_dirty");
+    let shape_dirty = ShapeDirty::new(sub_logger,on_dirty.clone());
+    let sub_logger  = logger.sub("mesh_registry_dirty");
+    let dirty_flag  = MeshRegistryDirty::new(sub_logger,on_dirty);
+    Self {shape_dirty,dirty_flag}
+}
+```
+
+
+### Vertical alignment
+
+The following elements should be aligned vertically in subsequent lines:
+- assignment operators (`=`),
+- type operators (`:`),
+- match arrows (`=>`), 
+- similar parameters or types.
+
+Examples: 
+```rust
+impl Printer for GlobalVarStorage {
+    fn print(&self, builder:&mut Builder) {
+        match self {
+            Self::ConstStorage      => build!(builder,"const"),
+            Self::UniformStorage    => build!(builder,"uniform"),
+            Self::InStorage  (qual) => build!(builder,"in" ,qual),
+            Self::OutStorage (qual) => build!(builder,"out",qual),
+        }
+    }
+}
+```
+
+
+### Spaces 
+- Type operator is not spaced: `fn test(foo:String, bar:Int) { ... }`.
+- Commas between comples expressions (including arg list) are spaced.
+- Commas between simple elements are not spaced: `Result<Self,Error>`.
+- Arguments to functions are not spaced: `build(builder,"out",qual)`.
```

Comment:
I personally think spaced args are more readable: `build(builder, "out", qual)`.

Reply:
I think it's easier to find groups of `id:type` if there is no space between them. So, for me, it's a preferable style when defining the function parameters in a single line.

Reply:
Guys, I would stay with the no-spaces rules here tbh. In Rust, we're using `,` to separate things. We use extra spacing when defining functions only to divide `something:Type` groups apart, but when calling a function there is no need for such operation. Using both `,` and a space is separating args like with two spaces. If you use color highlighting then its obvious where the `,` is. 

I recognize this (as the majority of this doc) as not very objective preference, so I'm just asking to stay with the current description here.